src/core/expression/qgsexpressionnodeimpl.cpp

```diff
@@ -91,6 +91,10 @@ QString QgsExpressionNode::NodeList::cleanNamedNodeName( const QString &name )
     cleaned = QStringLiteral( "geometry" );
   else if ( cleaned == QLatin1String( "val" ) )
     cleaned = QStringLiteral( "value" );
+  else if ( cleaned == QLatin1String( "geometry a" ) or cleaned == QLatin1String( "geometry_a" ) )
```

Comment:
```suggestion
  else if (  cleaned == QLatin1String( "geometry_a" ) )
```

The versions spaces in them would never have worked...